runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -418,6 +418,18 @@ public void stop() {
         }
     }
 
+    /**
+     * Reestablish the netty connection.
+     */
+    @Override
+    public void reconnect() {
+        // Close the channel and auto-trigger the reconnection callback.
+        log.info("NettyClientRouter reconnecting. Closing the existing channel.");
+        if (channel != null && channel.isOpen()) {
+            channel.close();
+        }
+    }
```

Comment:
I think you should capture the channel in a local variable because the global reference can change after the if-condition causing you to close a different channel than the one checked

Reply:
Done